anvio/migrations/modules/v3_to_v4.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python
+# -*- coding: utf-8
+
+import sys
+import argparse
+
+import anvio.db as db
+import anvio.utils as utils
+import anvio.terminal as terminal
+
+from anvio.errors import ConfigError
+
+
+current_version, next_version = [x[1:] for x in __name__.split('_to_')]
+
+brite_table_name      = "brite_hierarchies"
+brite_table_structure = ['hierarchy_accession', 'hierarchy_name', 'ortholog_accession', 'ortholog_name', 'categorization']
+brite_table_types     = [        'str'        ,       'str'     ,         'str'       ,      'str'     ,      'str'      ]
+
+run = terminal.Run()
+progress = terminal.Progress()
+
+def migrate(db_path):
+    if db_path is None:
+        raise ConfigError("No database path is given.")
+
+    utils.is_kegg_modules_db(db_path)
+
+    modules_db = db.DB(db_path, None, ignore_version = True)
+
+    if str(modules_db.get_version()) != current_version:
+        modules_db.disconnect()
+        raise ConfigError(f"Version of this modules database is not {current_version} (hence, this script cannot really do anything).")
+
+    progress.new("Migrating")
+    progress.update("...")
+
+    # just to be on the safe side.
+    try:
+        modules_db.drop_table(brite_table_name)
+    except:
+        pass
+
+    try:
+        modules_db.remove_meta_key_value_pair('num_brite_hierarchies')
+        modules_db.remove_meta_key_value_pair('total_brite_entries')
+    except:
+        pass
+
+    modules_db.set_meta_value('num_brite_hierarchies', None)
+    modules_db.set_meta_value('total_brite_hierarchies', None)
+
+    progress.update("Creating a new table for KEGG BRITE categorizations")
+
+    modules_db.create_table(brite_table_name, brite_table_structure, brite_table_types)
+
+    progress.update("Updating version")
+    modules_db.remove_meta_key_value_pair('version')
+    modules_db.set_version(next_version)
+
+    modules_db.disconnect()
+    progress.end()
+
+    run.info_single(f"The modules database is now {next_version}. An empty table of KEGG BRITE hierarchy "
+                    "categorizations of all orthologs was created, and related self table attributes were added. "
+                    "We suggest generating a new modules database from `anvi-setup-kegg-kofams` (using the `-D` flag "
+                    "in anvi'o v7 or earlier) and re-running `anvi-run-kegg-kofams` on your contigs databases "
```

Comment:
Shouldn't "anvi'o v7 or earlier" actually be "anvi'o v7 or later"? I don't think the -D flag exists prior to v7 :)

Reply:
I think I have clarified what I meant.